src/test/java/org/folio/rest/api/PrecedingSucceedingTitleTest.java

```diff
@@ -0,0 +1,297 @@
+package org.folio.rest.api;
+
+import static org.folio.rest.api.TestBaseWithInventoryUtil.UUID_ISBN;
+import static org.folio.rest.api.TestBaseWithInventoryUtil.identifier;
+import static org.folio.rest.support.http.InterfaceUrls.holdingsStorageUrl;
+import static org.folio.rest.support.http.InterfaceUrls.instancesStorageUrl;
+import static org.folio.rest.support.http.InterfaceUrls.itemsStorageUrl;
+import static org.folio.rest.support.http.InterfaceUrls.precedingSucceedingTitleUrl;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.folio.rest.api.entities.Instance;
+import org.folio.rest.api.entities.PrecedingSucceedingTitle;
+import org.folio.rest.jaxrs.model.Errors;
+import org.folio.rest.support.Response;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+public class PrecedingSucceedingTitleTest extends ResourceTestBase {
+
+  private static final String PRECEDING_SUCCEEDING_TITLE_TABLE = "preceding_succeeding_title";
+  private static final String TEST_TENANT = "test_tenant";
+  private static final String INSTANCE_TYPE_ID_TEXT = "6312d172-f0cf-40f6-b27d-9fa8feaf332f";
+  private static final String HRID = "inst000000000022";
+  private static final String TITLE = "A web primer";
+
+  @BeforeClass
+  public static void beforeAll() {
+    StorageTestSuite.deleteAll(TEST_TENANT, PRECEDING_SUCCEEDING_TITLE_TABLE);
+    StorageTestSuite.deleteAll(itemsStorageUrl(""));
+    StorageTestSuite.deleteAll(holdingsStorageUrl(""));
+    StorageTestSuite.deleteAll(instancesStorageUrl(""));
+  }
+
+  @Test
+  public void testPrecedingSucceedingTitleBasicCrud() throws MalformedURLException,
+    InterruptedException, TimeoutException, ExecutionException {
+
+    String entityPath = precedingSucceedingTitleUrl("").getPath();
+    JsonObject instance1Response = createInstance("Title One", INSTANCE_TYPE_ID_TEXT);
+    JsonObject instance2Response = createInstance("Title Two", INSTANCE_TYPE_ID_TEXT);
+    JsonArray identifiers = new JsonArray();
+    identifiers.add(identifier(UUID_ISBN, "9781473619777"));
+
+    PrecedingSucceedingTitle precedingSucceedingTitle = new PrecedingSucceedingTitle(
+      instance1Response.getString(Instance.ID_KEY), instance2Response.getString(Instance.ID_KEY),
+      TITLE, HRID, identifiers);
+    Response postResponse = createReferenceRecord(entityPath, precedingSucceedingTitle);
+    assertThat(postResponse.getStatusCode(), is(HttpURLConnection.HTTP_CREATED));
+
+    String entityUUID = postResponse.getJson().getString("id");
+
+    String updateProperty = PrecedingSucceedingTitle.TITLE_KEY;
+
+    testGetPutDeletePost(entityPath, entityUUID, precedingSucceedingTitle, updateProperty);
```

Comment:
Hi @Svitlana-Zmiivska, I'm not sure that the `preceding-succeeding` API falls under ReferenceRecordAPI term.

Usually `ReferenceRecordAPI` means a very basic CRUD API that used as a storage for static data (e.g. instance relationship types, identifiers types, locations, etc.).

Reply:
changed